### PR TITLE
Pica/Rasterizer: Add ETC1 texture decompression support.

### PR DESCRIPTION
**Pull request by archshift**
@neobrain's code from his pica_progress3 branch.

### CODE REVIEW

File: src/video_core/debug_utils/debug_utils.cpp [hunk 1]
```diff
@@ -441,6 +449,127 @@ const Math::Vec4<u8> LookupTexture(const u8* source, int x, int y, const Texture
...
+                                             + coarse_x * subtile_bytes * 4
+                                             + coarse_y * subtile_bytes * 4 * (info.width / 8)
+                                             + subtile_index * subtile_bytes * 8);
+        u64 alpha = 0xFFFFFFFFFFFFFFFF;
+        if (has_alpha) {
+            alpha = *source_ptr;
+            source_ptr++;
+        }
+
+        union {
```
Comment (neobrain) [inline src/video_core/debug_utils/debug_utils.cpp]: => `union ETC1Tile`

File: src/video_core/debug_utils/debug_utils.cpp [hunk 2]
```diff
@@ -441,6 +449,127 @@ const Math::Vec4<u8> LookupTexture(const u8* source, int x, int y, const Texture
...
+                if (GetNegationFlag(texel))
+                    modifier *= -1;
+
+                ret.r() = boost::algorithm::clamp(ret.r() + modifier, 0, 255);
+                ret.g() = boost::algorithm::clamp(ret.g() + modifier, 0, 255);
+                ret.b() = boost::algorithm::clamp(ret.b() + modifier, 0, 255);
+
+                return ret.Cast<u8>();
+            }
+        } const *etc1_tile = reinterpret_cast<decltype(etc1_tile)>(source_ptr);
```
Comment (neobrain) [inline src/video_core/debug_utils/debug_utils.cpp]: => drop the `decltype` thing and just use ETC1Tile (unless you think this is actually cleaner ;p)

**Comment by archshift** [comment]
Fixed.

File: src/video_core/debug_utils/debug_utils.cpp [hunk 3]
```diff
@@ -337,25 +338,30 @@ const Math::Vec4<u8> LookupTexture(const u8* source, int x, int y, const Texture
     i = (i ^ (i << 1)) & 0x1515;              // ---2 -1-0
     i = (i | (i >> 7)) & 0x3F;
 
-    source += coarse_y * info.stride;
-    const unsigned int offset = coarse_x * block_height + i;
+    if (info.format != Regs::TextureFormat::ETC1 &&
```
Comment (neobrain) [inline src/video_core/debug_utils/debug_utils.cpp]: Could you please add a TODO within the `if` branch:

    // TODO(neobrain): Fix code design to unify offsetting to the vertical block position!

**Comment by neobrain** [comment]
LGTM other than the missing TODO.

**Comment by archshift** [comment]
Good to go.

**Comment by bunnei** [comment]
LGTM

**Pull request merged**